File: 2016/7/7.js
"use strict";

const fs = require("fs");

let skattekart = fs.readFileSync("skattekart.txt", "utf-8");

console.time();

let pos = skattekart.split("\r\n").reduce((pos, line) => {
	line = line.split(" ");
	let length = Number.parseInt(line[1]);
	switch (line[3]) {
		case "north":
			pos[0] += length;
			break;
		case "east":
			pos[1] -= length;
			break;
		case "south":
			pos[0] -= length;
			break;
		case "west":
			pos[1] += length;
			break;
	}
	return pos;
}, [0, 0]);

console.log(pos);
console.timeEnd();

File: 2016/7/7.hs
import System.IO (readFile)

type Pos = (Int,Int)

walk :: Pos -> String -> Int -> Pos
walk (x,y) "north" m = (x,y+m)
walk (x,y) "south" m = (x,y-m)
walk (x,y) "east" m = (x-m,y)
walk (x,y) "west" m = (x+m,y)

foldFunc :: Pos -> String -> Pos
foldFunc pos line = walk pos direction (read meters)
    where [_, meters, _, direction] = words line

main = do
    contents <- readFile "skattekart.txt"
    let (x,y) = foldl foldFunc (0,0) $ lines contents
    putStrLn (show y ++ "," ++ show x)
